src/components/Image/ImageLoader.jsx

```diff
@@ -1,46 +1,141 @@
-import React, { Component } from 'react'
+import React, { PureComponent } from 'react'
 import { getDownloadLink } from 'cozy-client'
 
-export default class ImageLoader extends Component {
+const TTL = 6000
+
+const LOADING = 'LOADING'
+const LOADED = 'LOADED'
+const PENDING = 'PENDING'
+const FAILED = 'FAILED'
+
+export default class ImageLoader extends PureComponent {
   state = {
-    fallback: null
+    fallbackUrl: null,
+    primaryStatus: LOADING,
+    fallbackStatus: PENDING
+  }
+
+  isLoadingPrimary() {
+    return this.state.primaryStatus === LOADING
+  }
+  wantsToLoadPrimary() {
+    return this.isLoadingPrimary() && !this.img
+  }
+  isPrimaryLoaded() {
+    return this.state.primaryStatus === LOADED
+  }
+  isLoadingFallback() {
+    return (
+      this.state.primaryStatus === FAILED &&
+      this.state.fallbackStatus === LOADING
+    )
+  }
+  wantsToLoadFallbackUrl() {
+    return this.isLoadingFallback() && !this.state.fallbackUrl
+  }
+  wantsToLoadFallback() {
+    return this.isLoadingFallback() && this.state.fallbackUrl && !this.img
+  }
+  isFallbackLoaded() {
+    return this.state.fallbackStatus === LOADED
+  }
+  hasSomethingToShow() {
+    return this.isPrimaryLoaded() || this.isFallbackLoaded()
   }
```

Comment:
Tu déclines tous les status en fonction, j'ai l'impression qu'on pourrait peut-être avoir quelque chose de plus concis.
Peut-être on peut passer `this.state` en paramètre et composer des fonctions pures en dehors du composant : 

```js
const isFallbackloaded => ({fallbackStatus}) => fallbackStatus === LOADED
const isLoadingFallback => ({primaryStatus, fallbackStatus}) => primaryStatus === FAILED && fallbackStatus === LOADING
// etc.
```

Reply:
J'y ai pensé, et puis je me suis dit qu'on était de toute façon dans une classe donc dans de l'orienté objet, et que ça n'avait pas forcément de sens de chercher à extraire de la classe des choses comme ça, même si je te l'accorde, on pourrait gagner un peu en concision.

Reply:
Ça permettait aussi d'avoir des fonctions découplées de la classe permettant d'être facilement couvertes par des tests.